fetch/manip.go

```diff
@@ -83,7 +84,20 @@ func Resize(src io.Reader, c *CacheContext) (io.Reader, error) {
 	}
 
 	if c.Crop {
-		options.Height = c.Width
+		data := bytes.NewReader(raw)
+
+		image, _, err := image.Decode(data)
+		if err != nil {
+			return nil, err
+		}
+
+		minDimension := int(math.Min(float64(image.Bounds().Size().X), float64(image.Bounds().Size().Y)))
+
+		if minDimension < options.Width || options.Width == 0 {
```

Comment:
Bro way to leverage [the zero value](https://golang.org/ref/spec#The_zero_value)